etc/kernel-launchers/python/scripts/launch_ipykernel.py

```diff
@@ -360,7 +364,64 @@ def get_server_request(sock):
     return request_info
 
 
-def server_listener(sock, parent_pid):
+def get_application_id(ui_port):
+    application_id = ""
+    url = f"http://localhost:{ui_port}/api/v1/applications/"
+    response = requests.get(url)
+    if response.status_code == 200 and response.json():
+        application_id = response.json()[0].get("id")
+
+    return application_id
+
+
+def get_active_jobs_id(ui_port, application_id):
+    active_job_ids = []
+    url = "http://localhost:{}/api/v1/applications/{}/jobs?status=running".format(
+        ui_port, application_id
+    )
+    response = requests.get(url)
+    if response.status_code == 200 and response.json():
+        for job in response.json():
+            jobid = job.get("jobId")
+            active_job_ids.append(jobid)
+
+    return active_job_ids
+
+
+def kill_job(ui_port, jobid):
+    url = f"http://localhost:{ui_port}/jobs/job/kill/?id={jobid}"
+    response = requests.get(url)
+    if response.status_code == 200:
+        logger.info(f"Killed job with id: {jobid}")
+    else:
+        logger.warning(f"Failed to kill job with id: {jobid}")
```

Comment:
I think it would be a good idea to also include the applcationId (and kernel_id) in these messages so "paper trails" can be established...
Perhaps something like:
"Killed job '{job_id}' of application '{application_id}' and kernel '{kernel_id}'"